### ISSUE TITLE
Prometheus metrics enabled by default | how  to disable them 

### ISSUE THREAD

**Issue by ajitpalsinghjat** [cilium/cilium#31135]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

 Prometheus Metrics should not be enabled by default.  

Installing cilium with below command on k8s version 1.29  without any values yaml.

 #helm install cilium cilium/cilium --version 1.15.1 --namespace kube-system

### Cilium Version

v1.15.1

### Kernel Version

5.15.0-94-generic

### Kubernetes Version

1.29

### Regression

NA

### Relevant log output

```shell
root@i60408-A16124:~#
root@i60408-A16124:~# netstat -tunlp |grep 9962
tcp6       0      0 :::9962                 :::*                    LISTEN      95295/cilium-agent
```

### Anything else?

- we have to disable this port so it should be disable by default. 

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by joestringer** [comment]
Hi, thanks for the report. I've filed #31144 to revisit the decision to enable prometheus metrics by default. I agree that it is surprising behaviour that this is enabled by default, given that the default helm value for prometheus is `prometheus.enabled=false`. After a brief attempt to work around the issue using only Helm, I could not find a simple way to disable prometheus metrics in the agent without changing the Helm charts. 

As a workaround, you could try adding the following to the `cilium` ConfigMap and then restart all agents:

```
prometheus-serve-addr: ""
```

**Comment by ajitpalsinghjat** [comment]
Thanks for your quick response.

I have two queries :

1) Is there any future planning to disable this by default in up coming releases.  

2) why we are not able to add  " prometheus-serve-addr: ""   to cilium config map in helm charts itself. Is there any challenges as we are facing lot of security challenges as this port is open on every node and disabling this manually on each cluster is a challenge for us.

**Comment by joestringer** [comment]
For (1) it will be subject to the discussion on #31144 and whether it is backported. For (2) that's the bug, and it will require either daemon or helm changes to be able to allow the Helm charts to properly disable prometheus directly through Helm. I'd expect that whatever solution we come up with in order to allow you to disable prometheus, that fix will come in an upcoming monthly v1.15 patch release.

**Comment by ajitpalsinghjat** [comment]
Thank you for the update.
